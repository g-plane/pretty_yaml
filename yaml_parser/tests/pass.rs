use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
