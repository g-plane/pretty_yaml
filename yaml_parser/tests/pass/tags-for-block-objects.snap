---
source: yaml_parser/tests/pass.rs
---
ROOT@0..54
  DOCUMENT@0..53
    BLOCK@0..53
      BLOCK_MAP@0..53
        BLOCK_MAP_ENTRY@0..53
          BLOCK_MAP_KEY@0..3
            FLOW@0..3
              PLAIN_SCALAR@0..3 "foo"
          COLON@3..4 ":"
          WHITESPACE@4..5 " "
          BLOCK_MAP_VALUE@5..53
            BLOCK@5..53
              PROPERTIES@5..10
                TAG_PROPERTY@5..10
                  SHORTHAND_TAG@5..10
                    TAG_HANDLE@5..7
                      TAG_HANDLE_SECONDARY@5..7 "!!"
                    TAG_CHAR@7..10 "seq"
              WHITESPACE@10..13 "\n  "
              BLOCK_SEQ@13..53
                BLOCK_SEQ_ENTRY@13..22
                  MINUS@13..14 "-"
                  WHITESPACE@14..15 " "
                  FLOW@15..22
                    PROPERTIES@15..20
                      TAG_PROPERTY@15..20
                        SHORTHAND_TAG@15..20
                          TAG_HANDLE@15..17
                            TAG_HANDLE_SECONDARY@15..17 "!!"
                          TAG_CHAR@17..20 "str"
                    WHITESPACE@20..21 " "
                    PLAIN_SCALAR@21..22 "a"
                WHITESPACE@22..25 "\n  "
                BLOCK_SEQ_ENTRY@25..53
                  MINUS@25..26 "-"
                  WHITESPACE@26..27 " "
                  BLOCK@27..53
                    PROPERTIES@27..32
                      TAG_PROPERTY@27..32
                        SHORTHAND_TAG@27..32
                          TAG_HANDLE@27..29
                            TAG_HANDLE_SECONDARY@27..29 "!!"
                          TAG_CHAR@29..32 "map"
                    WHITESPACE@32..37 "\n    "
                    BLOCK_MAP@37..53
                      BLOCK_MAP_ENTRY@37..53
                        BLOCK_MAP_KEY@37..40
                          FLOW@37..40
                            PLAIN_SCALAR@37..40 "key"
                        COLON@40..41 ":"
                        WHITESPACE@41..42 " "
                        BLOCK_MAP_VALUE@42..53
                          FLOW@42..53
                            PROPERTIES@42..47
                              TAG_PROPERTY@42..47
                                SHORTHAND_TAG@42..47
                                  TAG_HANDLE@42..44
                                    TAG_HANDLE_SECONDARY@42..44 "!!"
                                  TAG_CHAR@44..47 "str"
                            WHITESPACE@47..48 " "
                            PLAIN_SCALAR@48..53 "value"
  WHITESPACE@53..54 "\n"
